low-level/files/opt/seagate/sspl/setup/sspl_setup_init.py

```diff
@@ -0,0 +1,181 @@
+#!/bin/env python3
+
+# Copyright (c) 2020 Seagate Technology LLC and/or its Affiliates
+#
+# This program is free software: you can redistribute it and/or modify it under the
+# terms of the GNU Affero General Public License as published by the Free Software
+# Foundation, either version 3 of the License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT ANY
+# WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+# PARTICULAR PURPOSE. See the GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>. For any questions
+# about this software or licensing, please email opensource@seagate.com or
+# cortx-questions@seagate.com.
+
+#################################################################
+# This script performs following operations.
+# - Creates datapath as defined in /etc/sspl.conf
+# - Check dependencies for consts.roles other than '<product>'
+#################################################################
+
+import sys
+import os
+import subprocess
+import pwd
+
+from cortx.sspl.lowlevel.framework.base.sspl_constants import file_store_config_path, roles
+from cortx.sspl.lowlevel.files.opt.seagate.sspl.setup.sspl_setup import Cmd as SSPLSetup
+
+import psutil
+
+class SetupInit:
+    """Init Setup Interface"""
+
+    name = "init"
+
+    SSU_DEPENDENCY_RPMS = [
+                "sg3_utils",
+                "gemhpi",
+                "pull_sea_logs",
+                "python-hpi",
+                "zabbix-agent-lib",
+                "zabbix-api-gescheit",
+                "zabbix-xrtx-lib",
+                "python-openhpi-baselib",
+                "zabbix-collector",
+    ]
+
+    SSU_REQUIRED_PROCESSES = [
+                    "openhpid",
+                    "dcs-collectord",
+    ]
+
+    VM_DEPENDENCY_RPMS = []
+
+    HPI_PATH = '/tmp/dcs/hpi'
+    MDADM_PATH = '/etc/mdadm.conf'
+
+    def __init__(self, args : list):
+        self.args = args
+        self.role = None
+
+    def usage(self):
+        sys.stderr.write(
+            f"{self.name} [check|create [-dp] [-r <ssu|gw|cmu|vm|cortx>]]\n"
+            "create options:\n"
+            "\t -dp Create configured datapath\n"
+            "\t -r  Role to be configured on the current node\n"
+            )
+        sys.exit(1)
+
+    def check_for_dep_rpms(self, rpm_list : list):
+        for dep in rpm_list:
+            name = SSPLSetup._send_command(f'rpm -q {dep}', fail_on_error=False)
+            if "b''" in name:
+                print(f"- Required rpm '{dep}' not installed, exiting")
+                sys.exit(1)
+    
+    def check_for_active_processes(self, process_list : list):
```

Comment:
Contribute this generic function to cortx-utils(@satish-seagate can you add this too in your PR), though until its available there, keep in sspl Utility.py and use from there until then to remain functional
check_for_active_processes()
check_for_dep_rpms()